### ISSUE TITLE
libm dependency problem

### ISSUE THREAD

**Issue by KrisThielemans** [conda-forge/parallelproj-feedstock#10]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Trying to build the STIR feedstock I get compilation problems. see https://github.com/conda-forge/stir-feedstock/pull/67#issuecomment-1419249922

This is likely caused by wrong dependencies. I've installed parallelproj via mamba on my Linux VM and checked `~/mambaforge/envs/parallelproj/lib/cmake/parallelproj/parallelprojTargets.cmake`. This has
```
# Create imported target parallelproj::parallelproj_c
add_library(parallelproj::parallelproj_c SHARED IMPORTED)

set_target_properties(parallelproj::parallelproj_c PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/"
  INTERFACE_LINK_LIBRARIES "/home/conda/feedstock_root/build_artifacts/parallelproj_1673649092245/_build_env/x86_64-conda-linux-gnu/sysroot/usr/lib/libm.so;OpenMP::OpenMP_C"
)
```
that seems wrong.

### Installed packages

```shell
parallelproj
```

### Environment info

```shell
active environment : parallelproj
    active env location : /home/sirfuser/mambaforge/envs/parallelproj
            shell level : 1
       user config file : /home/sirfuser/.condarc
 populated config files : /home/sirfuser/mambaforge/.condarc
                          /home/sirfuser/.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.10.6.final.0
       virtual packages : __linux=4.15.0=0
                          __glibc=2.27=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/sirfuser/mambaforge  (writable)
      conda av data dir : /home/sirfuser/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/sirfuser/mambaforge/pkgs
                          /home/sirfuser/.conda/pkgs
       envs directories : /home/sirfuser/mambaforge/envs
                          /home/sirfuser/.conda/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.10.6 Linux/4.15.0-29-generic ubuntu/18.04.2 glibc/2.27
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False
```

**Comment by gschramm** [comment]
Hi Kris,

already got feedback from conda-forge gitter.
It was pointed out, that in my cmake recipe, libm should be a private instead of a public dependency.
Do you think that would in principle solve the issue?
After reading the cmake docs, it makes sense to me that it should be a private dependency.

**Comment by KrisThielemans** [comment]
This goes beyond what I know TBH. I've never had problems with `libm` that I recall (although it vaguely runs a bell). In any case, should be easy to try: edit the `parallelprojTargets.cmake`- and remove the depdency, and see what STIR thinks. I can try that tomorrow.

**Comment by gschramm** [comment]
Removing ```libm``` from the ```INTERFACE_LINK_LIBRARIES``` in ```parallelprohTargets.cmake``` fixes the issue :)
I will release v1.2.14 where the ```libm``` dependency will be private

**Comment by gschramm** [comment]
Just released parallelproj v1.2.14. Should be on conda-forge by tomorrow (after the bot picks up the new version).

**Comment by KrisThielemans** [comment]
perfect. thanks!
